### DISCUSSION TITLE
v4.4.0

### DISCUSSION THREAD

**Discussion by jsjoeio** [coder/code-server#5172] [Announcements]
v4.4.0

VS Code v1.66.2

Upgrading is as easy as installing the new version over the old one. code-server
maintains all user data in `~/.local/share/code-server` so that it is preserved in between
installations.

### Changed

- Refactored methods in `Heart` class and made `Heart.beat()` async to make
  testing easier.
- Upgraded to Code 1.66.2.

### Added

- Added back telemetry patch which was removed in the Code reachitecture.
- Added support to use `true` for `CS_DISABLE_FILE_DOWNLOADS` environment
  variable. This means you can disable file downloads by setting
  `CS_DISABLE_FILE_DOWNLOADS` to `true` or `1`.
- Added tests for `Heart` class.

### Fixed

- Fixed installation issue in AUR after LICENSE rename.
- Fixed issue with listening on IPv6 addresses.
- Fixed issue with Docker publish action not being able to find artifacts. Now
  it downloads the release assets from the release.

## New Contributors
* @azriel-stephen made their first contribution in https://github.com/coder/code-server/pull/4981
* @cjtim made their first contribution in https://github.com/coder/code-server/pull/5045
* @dwahler made their first contribution in https://github.com/coder/code-server/pull/5068
* @rileycalhoun made their first contribution in https://github.com/coder/code-server/pull/5079
* @neilnaveen made their first contribution in https://github.com/coder/code-server/pull/5090
* @thx2001r made their first contribution in https://github.com/coder/code-server/pull/5078
* @awfulcooking made their first contribution in https://github.com/coder/code-server/pull/5133

**Full Changelog**: https://github.com/coder/code-server/compare/v4.1.0...v4.4.0

<hr /><em>This discussion was created from the release <a href='https://github.com/coder/code-server/releases/tag/v4.4.0'>v4.4.0</a>.</em>

**Comment by jsjoeio** [comment]
Homebrew: https://github.com/Homebrew/homebrew-core/pull/100983

**Comment by jsjoeio** [comment]
AUR: https://aur.archlinux.org/packages/code-server

**Comment by jsjoeio** [comment]
Docker: https://hub.docker.com/r/codercom/code-server/tags

**Comment by PeterBD0** [comment]
I can't install the french language pack, it says it's not installable for VS Code OSS but I was able to install it without any problems in version 3.10.2 ? What is the problem ?

**Reply by PeterBD0** [reply]
Wait I managed to install it, IDK what was the problem.

**Reply by jsjoeio** [reply]
I think it installs it on the server not the browser which may have lead to some confusion

**Reply by Dagherbou** [reply]
Oh wait a second, you are actually probably right. When I installed it it didin't save any of my 3.10.2 informations.

**Reply by Dagherbou** [reply]
I installed it on a ubuntu in termux by the way

**Reply by Dagherbou** [reply]
i guess yes then. Do you recommend the apk ?
